src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -346,6 +368,13 @@ private UnboundLambda BindAnonymousFunction(AnonymousFunctionExpressionSyntax sy
                     {
                         binder.ValidateLambdaParameterNameConflictsInScope(lambda.ParameterLocation(i), name, diagnostics);
                     }
+
+                    // An optional parameter cannot come before a required parameter
+                    if (i > 0 && lambda.ParameterHasDefault(i - 1) && !lambda.ParameterHasDefault(i))
```

Comment:
We'll need to talk about this a bit. I'm concerned with spreading these checks out more, as we already have a decent amount of duplication of the checks in `ParameterHelpers.ReportParameterErrors` (to our detriment, as we miss at least one error scenario from that method here).